### PR TITLE
refactor: container image lister as go binary, add support for reporting image size

### PR DESCRIPTION
**Pull request by cameronmeissner**
refactor: container image lister as go binary, add support for reporting image size

**What type of PR is this?**

**What this PR does / why we need it**:

refactors container image list (bom) generation during VHD builds to use a new go binary rather than an ad-hoc and error-prone bash script, also adds support for reporting the size (in bytes) of each listed container image

example diff (CBLMarinerV2Gen2ARM64): https://www.diffchecker.com/QiHqB0kW/ 

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:

```
none
```

### CODE REVIEW

File: vhdbuilder/lister/go.mod [hunk 1]
```diff
@@ -0,0 +1,62 @@
+module github.com/Azure/agentbaker/vhdbuilder/lister
+
+go 1.21
```
Comment (cameronmeissner) [inline vhdbuilder/lister/go.mod:3]: this makes me sad but the dev container we use doesn't support go 1.22...

File: packer.mk [hunk 2]
```diff
@@ -106,3 +106,13 @@ build-nbcparser-all:
 build-nbcparser-binary:
 	@echo "Building nbcparser binary"
 	@bash -c "pushd nbcparser && CGO_ENABLED=0 GOOS=linux GOARCH=$(ARCH) go build -o bin/nbcparser-$(ARCH) main.go && popd"
+
+build-lister-binary:
+ifeq (${ARCHITECTURE},X86_64)
+	@echo "Building lister binary for amd64"
+	@bash -c "pushd vhdbuilder/lister && CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o bin/lister main.go && popd"
```
Comment (djsly) [inline packer.mk:113]: could we use a variable for the GOARCH and only have one `go build` line ?

```
GOARCH=amd64
ifeq (${ARCHITECTURE},ARM64)
    GOARCH=arm64
endif
@echo "Building lister binary for ${GOARCH}"
@bash -c "pushd vhdbuilder/lister && CGO_ENABLED=0 GOOS=linux GOARCH=${GOARCH} go build -o bin/lister main.go && popd"
```

Comment (cameronmeissner) [reply packer.mk:113]: I tried that exact approach and I was having trouble getting it working, I can revisit though

Comment (cameronmeissner) [reply packer.mk]: refactored

**Pull request merged**